samcli/lib/cookiecutter/interactive_flow.py

```diff
@@ -0,0 +1,92 @@
+"""A flow of questions to be asked to the user in an interactive way."""
+from typing import Any, Dict, Optional
+import click
+from .question import Question
+
+
+class InteractiveFlow:
+    """
+    This class depends on the click module to prompt the user for answers to the questions of the flow.
+
+    Attributes
+    ----------
+    _questions: Dict[str, Question]
+        The questions to prompt the user for its answers
+    _first_question_key: str
+        The entry-point of the flow, it tells which question to ask the user first
+    """
+
+    def __init__(self, questions: Dict[str, Question], first_question_key: str):
+        self._questions: Dict[str, Question] = questions
+        self._first_question_key: str = first_question_key
+        self._current_question: Optional[Question] = None
+
+    @staticmethod
+    def _ask_a_question(question: Question) -> Any:
+        return click.prompt(text=question.text, default=question.default_answer)
+
+    @staticmethod
+    def _ask_a_yes_no_question(question: Question) -> bool:
+        return click.confirm(text=question.text, default=question.default_answer)
+
+    @staticmethod
+    def _ask_a_multiple_choice_question(question: Question) -> Any:
```

Comment:
`prompt_mulitple_choice` maybe?

Reply:
We have different type of questions, and each question has a different way to be asked. I feel like we should move the ask method to be part of the question itself, and we can have different types of Question class, and the asking logic can be included in each type .. so we can be open for adding new question type without needs to add new method here

Reply:
I thought about this too, but wanted to make the Question class unaware of "click". I might revise this later if I found this pattern is repeated again.